### ISSUE TITLE
Do newtypes need special handling?

### ISSUE THREAD

**Issue by RSWilli** [rudymatela/speculate#7]
I am using speculate to try to discover laws about Patterns in tidalcycles (https://hackage.haskell.org/package/tidal-1.7.1)

Since `Pattern a` is a data type similar to a function, I am using a newtype wrapper for which I implemented Listable, Eq and Show.

```haskell
import qualified Sound.Tidal.Context as T

instance Listable a => Listable (T.TPat a) where
  tiers =
    cons1 T.TPat_Seq
      \/ cons2 (\x y -> T.TPat_Elongate ((abs x + 1) % 1) y)
      \/ cons2 T.TPat_Atom

newtype MyPattern a = MyPattern {pat :: T.Pattern a}

instance (T.Enumerable a, T.Parseable a, Listable a) => Listable (MyPattern a) where
  list = map (MyPattern . T.toPat) list -- [T.TPat a] here

instance Eq a => Eq (MyPattern a) where
  MyPattern p1 == MyPattern p2 = T.queryArc p1 (T.Arc 0 1) == T.queryArc p2 (T.Arc 0 1)

instance Show a => Show (MyPattern a) where
  show (MyPattern p) = show $ T.queryArc p (T.Arc 0 1)
```

when I run speculate as following:

```haskell
main :: IO ()
main = do
  speculate
    args
      { constants =
          [ constant "fast" (fast :: MyPattern Rational -> MyPattern Int -> MyPattern Int),
            constant "slow" (slow :: MyPattern Rational -> MyPattern Int -> MyPattern Int)
          ]
      }
```

I get the following Warnings:

```
Warning: no Listable instance for MyPattern Int, variables of this type will not be considered
Warning: no Listable instance for MyPattern (Ratio Integer), variables of this type will not be considered
Warning: no Eq instance for MyPattern Int, equations of this type will not be considered
Warning: no Eq instance for MyPattern (Ratio Integer), equations of this type will not be considered
Warning: no Ord instance for MyPattern Int, inequations of this type will not be considered
Warning: no Ord instance for MyPattern (Ratio Integer), inequations of this type will not be considered
```

Are these Warnigns due to the newtype, or am I missing something completely different?

**Comment by jwaldmann** [comment]
Add `instances =...` (argument for `reify*` is any expression of the right type)
```
main = do
  speculate
    args
      { instances = [ reifyListable A, reifyEq A , reifyOrd A]
        , constants =
...

data T = A | B deriving (Eq, Ord, Show, Generic)
instance Listable T where tiers = genericTiers
```

see also https://github.com/rudymatela/speculate/blob/master/bench/nord.hs#L18

**Comment by RSWilli** [comment]
thanks that worked for me!

**Comment by jwaldmann** [comment]
see also #8 (improve documentation)

**Comment by rudymatela** [comment]
I've closed #8 now.

@RSWilli Here are the instructions on [how to deal with the `reifyInstances` family of functions](https://github.com/rudymatela/speculate#supported-types).
